### PR TITLE
Auth UI integration with email & polish

### PR DESCRIPTION
**Pull request by infomiho**
Integrating Auth UI with e-mail auth. It also adds error messages and success messages. `Auth.tsx` is bit big, but we'll cleanup afterwards 👍

### What was added
- `Auth.tsx` now has multiple states (login, signup, forgot password, reset password, email verify)
- For each of the steps we export a form e.g. `Login.tsx` or `ForgotPassword.tsx`
- Updates legacy social buttons to look the same as new ones
- Changes e-mail sending to be non-blocking

### CODE REVIEW

File: waspc/data/Generator/templates/server/src/auth/utils.ts [hunk 1]
```diff
@@ -137,7 +137,7 @@ async function sendEmailAndLogTimestamp(
   } catch (e) {
     rethrowError(e);  
   }
-  await emailSender.send(content);
+  emailSender.send(content);
```
Comment (infomiho) [inline waspc/data/Generator/templates/server/src/auth/utils.ts:140]: @shayneczyzewski after a  discussion with @matijaSos I've decided to remove the `await` here so we can avoid the 3 seconds loading in 99% and we'll just ignore the 1% failure cases for now

File: waspc/data/Generator/templates/react-app/src/auth/forms/Auth.tsx [hunk 2]
```diff
@@ -194,59 +178,108 @@ const SubmitButton = styled('button', {
...
-      window.alert(errorMessage(err))
+    },
+  });
+  {=/ isUsernameAndPasswordAuthEnabled =}
+  {=# isEmailAuthEnabled =}
+  const { handleSubmit, emailFieldVal, passwordFieldVal, setEmailFieldVal, setPasswordFieldVal } = useEmail({
+    isLogin,
+    onError: onErrorHandler,
+    showEmailVerificationPending() {
+      setSuccessMessage('Check your email for a confirmation link.')
```
Comment (matijaSos) [inline waspc/data/Generator/templates/react-app/src/auth/forms/Auth.tsx:243]: I'd enrich the message a bit, sth like "You've signed up successfully! Check your email ..."

**Review by matijaSos — APPROVED** [review]
LGTM! Didn't get into technical details, mostly UX.

**Review by shayneczyzewski — COMMENTED** [review]
First pass through code and playing with it looking really good @infomiho great job!!! I will do a bit more playing with existing apps with this then give it the ✅

**Review by shayneczyzewski — APPROVED** [review]
Did more testing with example apps and seems clean, nice work!! You can choose to do what you want with my comments. Approving.

Just one reminder for docs we will want to remind people the import has changed for components:

```sh
Web app(stderr):   ✘ [ERROR] No matching export in "src/auth/forms/Login.tsx" for import "default"
Web app(stderr):
Web app(stderr):     src/ext-src/LoginPage.jsx:3:7:
Web app(stderr):       3 │ import LoginForm from '../auth/forms/Login'
Web app(stderr):         ╵        ~~~~~~~~~
Web app(stderr):
Web app(stderr):
Web app(stderr): ✘ [ERROR] No matching export in "src/auth/forms/Signup.tsx" for import "default"
Web app(stderr):
Web app(stderr):     src/ext-src/SignupPage.jsx:3:7:
Web app(stderr):       3 │ import SignupForm from '../auth/forms/Signup'
Web app(stderr):         ╵        ~~~~~~~~~~
Web app(stderr):
```

Went from default to named. Since this release is so big we can call it breaking too and bump major if we want.

Nice job!! 🥳

File: waspc/data/Generator/templates/react-app/src/auth/forms/SocialIcons.tsx [hunk 3]
```diff
@@ -8,11 +8,11 @@ const defaultStyles = css({
 export const Google = () => (
   <svg
```
[the remark this answers is not in the document]
Comment (infomiho) [reply waspc/data/Generator/templates/react-app/src/auth/forms/SocialIcons.tsx:9]: `tsx` components play nicely with `jsx` components :) 

Users can import them like `import X from ./ComponentName` and not even care if it's TS or JS

File: waspc/src/Wasp/AppSpec/App/Auth.hs [hunk 4]
```diff
@@ -82,3 +79,8 @@ isGitHubAuthEnabled = isJust . gitHub . methods
 
 isEmailAuthEnabled :: Auth -> Bool
 isEmailAuthEnabled = isJust . email . methods
+
+isEmailVerificationRequired :: Auth -> Bool
+isEmailVerificationRequired auth = case email . methods $ auth of
+  Nothing -> False
+  Just emailAuthConfig -> not $ fromMaybe False (allowUnverifiedLogin emailAuthConfig)
```
[the remark this answers is not in the document]
Comment (infomiho) [reply waspc/src/Wasp/AppSpec/App/Auth.hs:86]: Yes sir, that's a nice improvement. Thanks!

File: waspc/src/Wasp/Generator/WebAppGenerator/AuthG.hs [hunk 5]
```diff
@@ -74,30 +79,24 @@ genAuthForm auth =
...
-    [ "onAuthSucceededRedirectTo" .= getOnAuthSucceededRedirectToOrDefault auth
-    ]
-
-genSignupForm :: AS.Auth.Auth -> Generator FileDraft
-genSignupForm auth =
-  compileTmplToSamePath
-    [relfile|auth/forms/Signup.jsx|]
-    [ "onAuthSucceededRedirectTo" .= getOnAuthSucceededRedirectToOrDefault auth
+      "gitHubSignInPath" .= OAuth.serverLoginUrl gitHubAuthProvider,
+      -- Username and password
```
[the remark this answers is not in the document]
Comment (infomiho) [reply waspc/src/Wasp/Generator/WebAppGenerator/AuthG.hs:91]: I want to group the variables a bit 🤷

File: waspc/data/Generator/templates/react-app/src/auth/forms/Auth.tsx (hunk 2 above)
Comment (infomiho) [reply waspc/data/Generator/templates/react-app/src/auth/forms/Auth.tsx:243]: I'll add it 👍

**Pull request merged**